### DISCUSSION TITLE
Is platform.findPr returns the oldest PR by design?

### DISCUSSION THREAD

**Discussion by Gabriel-Ladzaretti** [renovatebot/renovate#16803] [Request Help] [answered]
When sending the rest api request (github in this case) we explicitly request `sort=update direction=desc`.
meaning the the newest will show first -

In `getPrCache` at `lib/modules/platform/github/pr` we have:

<img width="1478" alt="image" src="https://user-images.githubusercontent.com/97394622/181217446-33972da9-91fb-4c89-ba35-6dc6eeebc745.png">

Which returns the same but reversed - 

<img width="1157" alt="image" src="https://user-images.githubusercontent.com/97394622/181218151-d26a0d3c-907e-4adf-b884-480cb4786517.png">

This all is expected as the keys used to index into the data are the PR numbers, the problems arises when using `platfom.findPr`
which will return the first entry it finds, which is the oldest.
this in general isn't a problem as a given branch shouldn't have too many PRs if any.
but for the edge case such as this - 
![image](https://user-images.githubusercontent.com/97394622/181219537-cd212bbf-0612-42f3-a108-05fe88301b8b.png)

The oldest pr will be returned and a new PR will be raised instead of reopening the autoclosed one.
<img width="1360" alt="image" src="https://user-images.githubusercontent.com/97394622/181219951-2d574289-4c99-4cf7-8ee5-d5b2b03845df.png">

As said, a given branch shouldn't have more than one PR in each state (open, closed etc.), but it might be a better to return the newest regardless.

To do so we can use `Array.prototype.findLast()` instead of current `Array.prototype.find()` in - 
https://github.com/renovatebot/renovate/blob/b1dd241c87fd589b0724ddb7132b41c7ca86064b/lib/modules/platform/github/index.ts#L626-L639

This holds for `github` as i dont know how we handle/key PRs in other platforms, but from what i saw we use `Array.prototype.find()` in all of them.

**Comment by viceice** [comment]
This is probably a bug in new caching behavior. @zharinov Can you look at it?

**Reply by Gabriel-Ladzaretti** [reply]
L90 is the source i think
https://github.com/renovatebot/renovate/blob/6698c206038b6b20e53f969fdac4bff3793400d9/lib/modules/platform/github/api-cache.ts#L74-L92

**Comment by zharinov** [comment, marked as answer]
The sorting order isn't guaranteed, however maybe it's a good idea to sort items before assigning to `config.prList`

**Reply by rarkins** [reply]
As long as it doesn't interfere with the caching

**Reply by zharinov** [reply]
I'll refactor it

**Reply by nabeelsaabna** [reply]
convert to issue ?

**Reply by zharinov** [reply]
Yes, please
